### DISCUSSION TITLE
Ability to run just the parts of the dependency graph which are ancestors to some project

### DISCUSSION THREAD

**Discussion by Nokel81** [nrwl/nx#23040] [Feature Requests]
**Background:**
I have a decently complex project graph (>250 projects) so my team is very cognizant of the build times during development. One thing that I think would help a lot would be a smarter `nx run-many` which could be used in conjunction with `nx watch` to do even less work.

**Proposal:**
Add the flag `--inverted` to `run-many` which inverts the graph and only runs the targets for the specified projects and those projects which depend on them.

Example:

```
nx watch -p "@my-scope/*" -- nx run-many --inverted -t "build,build:types" -p \$NX_PROJECT_NAME
```

Given a rough project graph similar to:
```json
{
  "nodes": [
    "@my-scope/foo",
    "@my-scope/bar",
    "@my-scope/blat",
    "@my-scope/ding",
    "@my-scope/woot"
  ],
  "dependencies": {
    "@my-scope/foo": [
      "@my-scope/woot",
      "@my-scope/ding",
      "@my-scope/blat",
      "@my-scope/bar"
    ],
    "@my-scope/bar": [
      "@my-scope/woot",
      "@my-scope/ding",
      "@my-scope/blat"
    ],
    "@my-scope/blat": [
      "@my-scope/woot",
      "@my-scope/ding"
    ],
    "@my-scope/ding": [
      "@my-scope/woot"
    ],
    "@my-scope/woot": []
  }
}
```

Then if the package `"@my-scope/woot"` were to be changed then all packages get rebuilt, but if package `"@my-scope/bar"` were to be changed only `"@my-scope/bar"` and `"@my-scope/foo"` would have their targets run (modulo caching)

**Comment by jaysoo** [comment]
@Nokel81 What type of projects is your team developing? Trying to understand the use case here, as it sounds like you are rebuilding projects often, and this is a way to only rebuild what's affected? 

What would the before and after look like for your team with and without this feature?

**Reply by Nokel81** [reply]
Yes this is part of a local build workflow. We are developing a large desktop application and use packages as a means of splitting the work. 

With this feature we would hopefully massively speed up the development cycle by doing less work and by having more trust in our build system. Since we would want a change that affects the types of a package to cause the system to inform the developer which other packages need to updated.

**Reply by jaysoo** [reply]
Is there a reason you cannot run the build for all apps, and rely on caching to speed things up?

**Reply by Nokel81** [reply]
We tried that but maybe our configuration is wrong since nothing rebuilds. We have to touch a file in a different package for nx to actually rebuild package. FYI we are currently using lerna@v6 because we have noticed about a 50% degradation in performance when moving to a newer version of lerna (or even nx)
